### PR TITLE
Multihop: enable stable pool options

### PR DESCRIPTION
**Pull request by gangov**

### CODE REVIEW

**Review by ueco-jb — COMMENTED** [review]
Alright, idea with straight up copying messages was a bad one. Sorry.
We'll expand the interface and introduce the mandatory PoolType swap to a path.

**Review by ueco-jb — COMMENTED** [review]
Small optimization plus the missing MIGRATION.md entry.

File: contracts/multihop/src/contract.rs [hunk 1]
```diff
@@ -168,25 +219,45 @@ impl MultihopTrait for Multihop {
...
+                        .commission_amounts
+                        .push_back((token_symbol, simulated_reverse_swap.commission_amount));
+                    simulate_swap_response.offer_amount = simulated_reverse_swap.offer_amount;
+                    simulate_swap_response
+                        .spread_amount
+                        .push_back(simulated_reverse_swap.spread_amount);
+
+                    next_ask_amount = simulated_reverse_swap.offer_amount;
+                }
+            }
```
[the remark this answers is not in the document]
Comment (gangov) [reply contracts/multihop/src/contract.rs:260]: this is what I wanted to do yesterday, but unfortunately it fails with:

```Rust
`match` arms have incompatible types
`SimulateReverseSwapResponse` and `SimulateReverseSwapResponse` have similar names, but are actually distinct types

lib.rs(19, 5): `SimulateReverseSwapResponse` is defined in module `crate::stable_pool` of the current crate
lib.rs(12, 5): `SimulateReverseSwapResponse` is defined in module `crate::xyk_pool` of the current crate
```

[the remark this answers is not in the document]
Comment (gangov) [reply contracts/multihop/src/contract.rs:260]: keeping it as is, because of that

**Review by ueco-jb — APPROVED** [review]
LGTM

**Pull request merged**